src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1122,38 +1125,214 @@ class spell_pri_power_word_shield : public SpellScriptLoader
+                InjuredPlayers,
+                InjuredPets,
+                FullHealthPlayers,
+                None,
+            };
+
+            void OnTargetSelect(std::list<WorldObject*>& targets)
+            {
+                // find best target
+                // Make sure too exclude the player that had the aura (caster in our case)
+                Unit* owner = GetCaster();
+                TargetPriority priority = TargetPriority::None;
+                for (auto iter = targets.cbegin(); iter != targets.cend(); ++iter)
                 {
-                    if (AuraEffect* aurEff = caster->GetAuraEffect(SPELL_PRIEST_T9_HEALING_2P, EFFECT_0))
+                    WorldObject const* worldObject = *iter;
+                    if (!worldObject || worldObject == owner)
+                        continue;
+                    if (Player const* player = worldObject->ToPlayer())
                     {
-                        int32 heal = GetHitHeal();
-                        AddPct(heal, aurEff->GetAmount());
-                        SetHitHeal(heal);
+                        if (!player->IsFullHealth())
+                        {
+                            priority = TargetPriority::InjuredPlayers; // highest priority, no need to check anything else
+                            break;
+                        }
+                        if (priority == TargetPriority::None)
+                            priority = TargetPriority::FullHealthPlayers;
                     }
+                    else if (Creature const* creature = worldObject->ToCreature())
+                        if (!creature->IsFullHealth())
+                            priority = TargetPriority::InjuredPets;
                 }
+                if (priority == TargetPriority::None)
+                    targets.clear();
+                else
+                {
+                    targets.remove_if([priority](WorldObject* worldObject)
+                    {
+                        if (!worldObject)
+                            return true;
+
+                        switch (priority)
+                        {
+                            case TargetPriority::InjuredPlayers: return !(worldObject->ToPlayer()) || worldObject->ToPlayer()->IsFullHealth();
+                            case TargetPriority::InjuredPets: return !(worldObject->ToCreature()) || worldObject->ToPlayer();
+                            case TargetPriority::FullHealthPlayers: return !(worldObject->ToPlayer());
+                            default: return true;
+                        }
+                    });
+
+                    // choose one random target from targets
+                    if (targets.size() > 1)
+                    {
+                        uint32 chosen = urand(0, targets.size() - 1);
+                        uint32 idx = 0;
+                        targets.remove_if([chosen, &idx](WorldObject* worldObject) { return (chosen != idx++); });
```

Comment:
Let's hope this line is the last one for now (unused parameter).
```suggestion
                        targets.remove_if([chosen, &idx](WorldObject* /*worldObject*/) { return (chosen != idx++); });
```

Reply:
yeah I saw that. Was trying to make it show up on my VS with cmake flags but couldn't; just did it manually, sadly.

Reply:
I know. It can be frustrating to deal with unused parameters/arguments reported by the checks (Travis & ci/circleci) since they stop checking the rest of the code when one is found.

Reply:
There is a better way to do it
```
WorldObject* selected = Trinity::Containers::SelectRandomContainerElement(targets);
targets.clear();
targets.push_back(selected);
```